control/Director.php

```diff
@@ -889,7 +890,11 @@ public static function forceSSL($patterns = null, $secureDomain = null) {
 	 * Force a redirect to a domain starting with "www."
 	 */
 	public static function forceWWW() {
-		if(!Director::isDev() && !Director::isTest() && strpos($_SERVER['HTTP_HOST'], 'www') !== 0) {
+		if(!Director::is_cli() 
+			&& !Director::isDev() 
```

Comment:
> The environment type could be set to 'blah'.

It *could* be, but that would be wrong. `Director::get_environment_type()` will just return false if it's not one of the 3 expected types.

---

I don't see why this function is dictating the environments this should work on. It's not `forceSSLIfNotTestOrDev`. The developer should make the call if they want to force www on the test or live environments.

Having functionality that only works on one stage can hide bugs and cause unexpected and un-testable results.